src/complex/Utilities/OStreamUtilities.hpp

```diff
@@ -0,0 +1,765 @@
+            sstrm << "\n";
+          }
+        }
+      }
+      else // no delimiter / check binary as binary should NEVER uses delimiter
+      {
+        if(m_IsBinary)
+        {
+          for(size_t i = start; i < end; i++)
+          {
+            sstrm << m_Matrix.getValue(i, m_VertColumn);
+          }
+        }
+        else
+        {
+          for(size_t i = start; i < end; i++)
+          {
+            sstrm << m_Matrix.getValue(i, m_VertColumn);
+            if(i != 0 && i % length == 0)
+            {
+              sstrm << "\n";
+            }
+          }
+        }
+      }
+    }
+    else // unbalanced data arrays (must search for uninitialized)
+    {
+      if(start > length) // figure out position relative to current column
+      {
+        count = static_cast<size_t>(start % length);
+      }
+      else
+      {
+        count = start;
+      }
+      if(!m_Delim.empty())
+      {
+        for(size_t i = start; i < end; i++)
+        {
+          const auto& selected = m_Matrix.getValue(i, m_VertColumn);
+          if(selected.find("UNINITIALIZED") == std::string::npos) // if is not empty insert
+          {
+            sstrm << selected << m_Delim;
+            count++;
+            if(count == length)
+            {
+              sstrm << "\n";
+              count = 0;
+            }
+            else if(i + 1 < highestIndex) // if next is less than max
+            {
+              if(m_Matrix.getValue(i + 1).find("UNINITIALIZED") != std::string::npos) // if next is empty
+              {
+                sstrm << "\n";
+                count = 0;
+              }
+            }
+          }
+        }
+      }
+      else // empty delimiter / check binary as binary should NEVER uses delimiter
+      {
+        if(m_IsBinary)
+        {
+          for(size_t i = start; i < end; i++)
+          {
+            const auto& selected = m_Matrix.getValue(i, m_VertColumn);
+            if(selected.find("UNINITIALIZED") == std::string::npos) // if is not empty insert
+            {
+              sstrm << selected;
+            }
+          }
+        }
+        else
+        {
+          for(size_t i = start; i < end; i++)
+          {
+            const auto& selected = m_Matrix.getValue(i, m_VertColumn);
+            if(selected.find("UNINITIALIZED") == std::string::npos) // if has value
+            {
+              sstrm << selected;
+              count++;
+              if(count == length)
+              {
+                sstrm << "\n";
+                count = 0;
+              }
+              else if(i + 1 < highestIndex) // if next is less than max
+              {
+                if(m_Matrix.getValue(i + 1).find("UNINITIALIZED") != std::string::npos) // if next is empty
+                {
+                  sstrm << "\n";
+                  count = 0;
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    m_StringStore[start] = std::move(sstrm.str());
```

Comment:
```suggestion
    m_StringStore[start] = sstrm.str();
```
`std::stringstream::str()` returns a new `std::string` so `std::move` is not required.